### PR TITLE
Configured css theme

### PR DESCRIPTION
**Pull request by bellons91**

### CODE REVIEW

File: public/mix-manifest.json
[the remark this answers is not in the document]
Comment (bellons91) [reply public/mix-manifest.json:1]: Ok, I removed this folder.

**Comment by igor-baiborodine** [comment]
> I should probably fork your repository, customize it as I want, and then add it as a submodule (so, replacing the value of url seen above with my forked repository.

I would rather start from scratch instead of copying the Bilberry Sandbox repository since I'm currently using it to test the `v4` version. Have you seen this [guide](https://www.kiroule.com/article/start-blogging-with-github-hugo-and-netlify/)? Just empty your repository and follow the guide. Try to deploy to Netlify a vanilla website. Then do the customizations you want.

**Comment by bellons91** [comment]
> Try to deploy to Netlify a vanilla website. Then do the customizations you want

Creating and deploying the empty site worked correctly.

But when I try to customize the theme, and I run `npm run dev` (after an `npm install`) I get
> The system cannot find the path specified.

**Comment by bellons91** [comment]
Apparently, I'm almost there.

Your solution worked well locally, but not on Netlify. 

And, since I can't push my changes to your repository, I performed the following steps:

1. fork your [theme](https://github.com/Lednerb/bilberry-hugo-theme), not the sandbox
2. add it as a submodule in my _themes_ folder:
2.1. `cd themes`
2.2. `git submodule add  https://github.com/code4it-dev/bilberry-hugo-theme.git`
3. Update the theme:
3.1 `cd themes\bilberry-hugo-theme`
3.2 `npm install`
3.3 change CSS style
3.4 `git add . && git commit -m "message" && git push`
3.5 `npm run production`
4 run the project locally:
4.1 `cd ..\..\`, so, on the root of the project
4.2: `hugo server`

now everything works correctly and the build does not break.
**Important: I had to upgrade my local GIT version: the older one had some issues with dirty submodules: by installing the latest version, the error disappeared**

Now: the build compiles, but I still can't see any CSS changes.

I suppose that's because Netlify does not know that it must run `npm run production` on the submodule to build the new style.

Am I on the right path?

**Comment by bellons91** [comment]
> I can't push my changes to your repository
because I want to customize the theme and some other pages, like the ~/page/2/

**Comment by igor-baiborodine** [comment]
> 3.3 change CSS style
3.4 git add . && git commit -m "message" && git push
3.5 npm run production

Here is the correct sequence:
3.3 change CSS style
3.4 npm run production
3.5 git add . && git commit -m "message" && git push

**Comment by bellons91** [comment]
That's odd, it still does not work. It works locally, but when it comes to the deployment, I cannot see my changes.

Let me share some more info about the steps I follow - maybe it's just a matter of git commits.

**Git status**: no changes

## change CSS style

I update the _variables.scss file:
image

## npm run production

The script runs correctly:
image

I can see changes in both the main project and the theme:

**Theme:**
image

**Blog**:
image

A detail of the changes in _bilberry-hugo-theme_: it now references a `-dirty` commit
image

## Commit and push Theme changes

run `git add . && git commit -m "message" && git push`

image

by looking at the details of the last commit, I can see that everything was committed correctly:
image

The ID of the new commit is _a3d23823dd6d3957b518d65f4836728ea0235d8b_

## Run the project locally

Head back to the root folder.
It now still contains a changed file (see above: it's the one with the dirty commit ID)

**Note: when pushing the changes to our theme, we created a commit ID _a3d23823dd6d3957b518d65f4836728ea0235d8b_, but now it looks like the project is referencing another commit id, _be15a48d733a922943b4be2df1473837979596a3-dirty_. Maybe here's the cause of the error.**

Now, run `hugo server`.
(continue...)

**Comment by bellons91** [comment]
Hugo server worked correctly:
image

And the theme loads the awful color I set in the _variables file:

image

## Push the changes
again, run ` git add . && git commit -m "message" && git push`.

and here we are: https://github.com/code4it-dev/c4it-hugo/pull/4/commits/ae2ba69fd437ffe1ca1950403154f826f735331a

**Comment by bellons91** [comment]
The site is then built and deployed correctly on Netlify ( https://app.netlify.com/sites/incredible-snickerdoodle-95e2af/deploys/642e8588163381000850845a )

but the theme is not being applied

https://642e8588163381000850845a--incredible-snickerdoodle-95e2af.netlify.app/

image

**Comment by igor-baiborodine** [comment]
I suspect that your website submodule is not in sync with the fork. In site's root execute **cd themes/bilberry-hugo-y-theme**, then **git log**. You should see the changes you made in the fork. If it's not the case do the Git submodule syncing. Check this article:

https://www.kiroule.com/article/simplify-development-and-testing-with-bilberry-sandbox/

**Comment by bellons91** [comment]
@igor-baiborodine, I found the solution (at least for the deployment preview)!

I was missing two parts:
1. the _netlify.toml_ file, which I copied from your sandbox
2. the definition of the branch used in the module

You can see the updated files here
https://github.com/code4it-dev/c4it-hugo/pull/4/commits/27dd0fb437f52b5e454e8a878c51f83c7d2b7895

Also, in a previous commit, [I changed the repository name in the package-json file of the theme](https://github.com/code4it-dev/bilberry-hugo-theme/commit/91316bd916351fcb546fd453382c5b2df5b59d9f) (I have no idea if it solved some issues, though).

Now, let's see if the deployment on Master works well.

However, I suppose I'll not be the only developer who will face this error - so feel free to reference this discussion in your theme's README file

**Comment by igor-baiborodine** [comment]
@bellons91, thanks. I linked this thread to the original issue.

**Pull request merged**